model/flow/header.go

```diff
@@ -43,6 +43,32 @@ type Header struct {
 	// signature since the data represents cryptographic signatures serialized in some way (concatenation or other)
 }
 
+func NewHeader(
+	chainID ChainID,
+	parentID Identifier,
+	height uint64,
+	payloadHash Identifier,
+	timestamp time.Time,
+	view uint64,
+	parentVoterIndices []byte,
+	parentVoterSigData []byte,
+	proposerID Identifier,
+	proposerSigData []byte) *Header {
```

Comment:
Maybe to simplify migration that can be a single `HeaderParams` argument?  (It will also slightly improve usage since the user would not need to explicitly mention all the fields and they would also have names.)